src/main/scala/org/ergoplatform/nodeView/state/UtxoState.scala

```diff
@@ -146,44 +143,44 @@ class UtxoState(override val version: VersionTag,
 
 object UtxoState {
   private lazy val bestVersionKey = Algos.hash("best state version")
-  val EmissionBoxKey = Algos.hash("emission box jey")
+  val EmissionBoxIdKey = Algos.hash("emission box id key")
 
   private def metadata(modId: VersionTag,
                        stateRoot: ADDigest,
-                       emissionBoxOpt: Option[ErgoBox]): Seq[(Array[Byte], Array[Byte])] = {
+                       currentEmissionBoxOpt: Option[ErgoBox],
+                       context: ErgoStateContext): Seq[(Array[Byte], Array[Byte])] = {
     val idStateDigestIdxElem: (Array[Byte], Array[Byte]) = modId -> stateRoot
     val stateDigestIdIdxElem = Algos.hash(stateRoot) -> modId
     val bestVersion = bestVersionKey -> modId
-    val eb = EmissionBoxKey -> emissionBoxOpt.map(emissionBox => ErgoBoxSerializer.toBytes(emissionBox))
-      .getOrElse(Array.empty)
+    val eb = EmissionBoxIdKey -> currentEmissionBoxOpt.map(emissionBox => emissionBox.id).getOrElse(Array[Byte]())
+    val cb = ErgoStateReader.ContextKey -> context.bytes
 
-    Seq(idStateDigestIdxElem, stateDigestIdIdxElem, bestVersion, eb)
+    Seq(idStateDigestIdxElem, stateDigestIdIdxElem, bestVersion, eb, cb)
   }
 
-  def create(dir: File, emission: CoinsEmission, nodeViewHolderRef: Option[ActorRef]): UtxoState = {
-    val store = new LSMStore(dir, keepVersions = ErgoState.KeepVersions)
+  def create(dir: File, constants: StateConstants): UtxoState = {
```

Comment:
Why not to move storage folder to StateConstants as well?

Reply:
Because we do not need it inside a state, `StateConstants` just keeps fields, required for state, that do not change with the time.